### ISSUE TITLE
Loading isodate after biblatex causes MakeSentenceCase* to misbehave

### ISSUE THREAD

**Issue by alex-ball** [plk/biblatex#1063]
In v3.15a, loading the isodate package after biblatex means MakeSentenceCase* only makes a correct calculation about whether the current language is in the DeclareCaseLangs list if the language of a given entry is explicitly specified with `langid`.

MWE:

```latex
\documentclass[varwidth=67ex]{standalone}
\usepackage[british]{babel}
\usepackage{biblatex}
\addbibresource{biblatex-examples.bib}
\DeclareFieldFormat{titlecase}{\MakeSentenceCase*{#1}}
\usepackage[british,cleanlook]{isodate}

\begin{document}
Literal language british: \ifcaselang[british]{True}{False}.\\
\verb|\currentlang| \currentlang: \ifcaselang[\currentlang]{True}{False}.\\
\verb|\blx@languagename| \makeatletter\blx@languagename: \ifcaselang[\blx@languagename]{True}{False}.\makeatother\\
Implicit \verb|\currentlang|: \ifcaselang{True}{False}.\\

\MakeSentenceCase*{In Sentence Case} \cite{shore}.

\printbibliography
\end{document}
```

(The `shore` reference has Title Case in the bib file but no `langid` set.)

Expected output:

```markdown
Literal language british: True.
\currentlang british: True.
\blx@languagename british: True.
Implicit \currentlang: True.
In sentence case [1].

**References**

[1] Bradd Shore. “Twice-born, once conceived. Meaning construc-
tion and cultural cognition”. In: *American anthropologist.* New
ser. 93.1 (Mar. 1991), pp. 9–27.
```

I have done some testing with different versions; in all cases `\ifcaselang[british]` comes out True.

v3.15a - isodate before biblatex:

- Other `\ifcaselang` tests come out True
- `\MakeSentenceCase*` works as expected
- Titles in bibliography print as expected

v3.15a - isodate after biblatex:

- Other `\ifcaselang` tests come out False
- `\MakeSentenceCase*` does not change case
- Titles in bibliography print in Title Case

v3.14 - isodate before biblatex:

- Other `\ifcaselang` tests come out True
- `\MakeSentenceCase*` works as expected
- Titles in bibliography print as expected

v3.14 - isodate after biblatex:

- Other `\ifcaselang` tests come out False
- `\MakeSentenceCase*` works as expected
- Titles in bibliography print as expected

v3.13a - isodate before biblatex:

- Other `\ifcaselang` tests come out False
- `\MakeSentenceCase*` works as expected
- Titles in bibliography print as expected

v3.13a - isodate after biblatex:

- Other `\ifcaselang` tests come out False
- `\MakeSentenceCase*` works as expected
- Titles in bibliography print as expected

The results are the same regardless of whether expl3 or latex2e syntax is used.

I am not sure what it is about isodate that clobbers the more recent definitions of `ifcaselang`, so if avoiding the clash is not desirable, should there be a note in the documentation about the incompatibility?

**Comment by moewew** [comment]
I think this is the same bug as https://github.com/plk/biblatex/issues/1057.

If I'm right the problem is that `isodate` does something like
```latex
\AtBeginDocument{%
  \if[babel loaded]
    \gdef\iso@languagename{\languagename}%
    \edef\iso@tmplang{\languagename}%
    \expandafter\selectlanguage\expandafter{\iso@tmplang}%
  \fi
}
```
This then means that `\languagename` is set by `\selectlanguage` (i.e. by `babel`) and gets a weird catcode regime, which throws off our new code.

https://github.com/plk/biblatex/commit/b08d5cdd38e99cd8fbe05133d8d4c8f969b6f358 for #1057 tries to work around the catcode weirdness by `\detokenize`ing everything.

**Comment by moewew** [comment]
Since the MWE compiles fine with the fix for #1057 applied (in particular https://github.com/plk/biblatex/commit/b08d5cdd38e99cd8fbe05133d8d4c8f969b6f358) I'm labelling this 'fixedindev'.

Of course further tests with the `dev` branch would be appreciated.

**Comment by moewew** [comment]
`biblatex` v3.16 with the fix for #1057 was released on 31 Dec 2020 and is available now in TeX live and MikTeX. AFAICS, the MWE from here compiles fine now.
